packages/next/types/index.d.ts

```diff
@@ -12,6 +12,7 @@ import {
 } from 'next-server/dist/lib/utils'
 
 import { PageConfig } from 'next-server/types'
+import { RouterUrl } from 'next-server/dist/lib/router/router'
```

Comment:
Should this be `import { PageConfig, RouterUrl } from 'next-server/types'` with `RouterUrl` re-exported in `next-server/types/index.d.ds`? Not sure if it's worth it, just asking 🙂

Reply:
Hmm, that is a good point, I think we will change everything to have it only exported from `next-server/types` :)

Reply:
Main `README` does not mention `next-server` anywhere, which makes it an internal package about which the end user does not necessary know about. I'd be in favour of seeing all types re-exported in a single entry point, so that there was just one import statement: `import { Foo, Bar, Baz } from 'next'`.

Similarly, default exports from `next/router` and `next/link` could be rewired one day so that extra entry points went deprecated and eventually removed. Grabbing _all_ exports from the module's index makes it easy to maintain typings and stuff, while still leaving things tree-shakeable. IMHO a user should not know anything about the internal folder structure of a package, it's just one 'file'.

Reply:
@kachkaev `next-server` is never mentioned because we only want people to use `next`, getting rid of `next/router`, `next/amp`, `next/link` and e.t.c may be good but it also has disadvantages. We have a modular setup that only requires you to import what you need, many projects don't require a lot of the modules we have, like `amp`, and it makes sense to add the types where the modules live.